apps/rxjs/11-high-order-operator-bug/src/app/app.service.ts

```diff
@@ -12,11 +12,9 @@ export class AppService {
     return this.dbService.searchByType(type).pipe(
       take(1),
       mergeMap((topicToDelete) =>
-        topicToDelete.length > 0
-          ? topicToDelete
-              .map((t) => this.dbService.deleteOneTopic(t.id))
-              .reduce((acc, curr) => merge(acc, curr), of(true))
-          : of(true),
+        forkJoin(
+          topicToDelete.map((t) => this.dbService.deleteOneTopic(t.id)),
```

Comment:
forkJoin is one solution, you can also use mergeAll and toArray. I find the latest a bit easier to read but both works the same